### ISSUE TITLE
How to provide 2 or more images to the model and compare the beauty of the images?

### ISSUE THREAD

**Issue by xinsir6** [haotian-liu/LLaVA#874]
### Question

I have a problem when use llava to process multi images in the same time, such as, give the model 2 or more images, and ask it to answer questions about the images, like which one do you like better? The web demo and chat can't solve this problem, so could you provide a special scripts to do this?

**Comment by mapluisch** [comment]
You might try putting two images into one by separating them with a horizontal line and including that detail in your prompt. I haven't tested this method extensively, but it seemed to work quite well in a quick trial I did: 

<img width="1073" alt="fused_image" src="https://github.com/haotian-liu/LLaVA/assets/31780571/718f2cac-b0d6-4e92-9f61-0bd0e89679b1">

If this works well enough for you, you could adjust the scripts to accept two images which then get merged in this way.

**Comment by xinsir6** [comment]
good idea, I will try it, the only problem is that the method requires two images to be resized into same width/height

**Comment by mapluisch** [comment]
> good idea, I will try it, the only problem is that the method requires two images to be resized into same width/height

I've made a simple proof of concept for you based on cli.py:

https://github.com/mapluisch/LLaVA-CLI-with-multiple-images

It doesn't resize (or truncate / crop) the images, just concatenates them vertically.

**Comment by xinsir6** [comment]
> > good idea, I will try it, the only problem is that the method requires two images to be resized into same width/height
> 
> I've made a simple proof of concept for you based on cli.py:
> 
> https://github.com/mapluisch/LLaVA-CLI-with-multiple-images
> 
> It doesn't resize (or truncate / crop) the images, just concatenates them vertically.

![image](https://github.com/haotian-liu/LLaVA/assets/128811208/a02af74c-7d28-4781-9c29-7804e7a245da)
![image](https://github.com/haotian-liu/LLaVA/assets/128811208/b922f363-24c9-439d-8190-4fb7db7abeca)
I try to ask the model as the way you request, but the model refuse to reply, is there any way to solve it ？

**Comment by mapluisch** [comment]
You should play around with different prompts and temperatures. I'm using 4-bit quantization on the 13b model, and this prompt works: 

`Analyze the two images and tell me which one is better and why`

**Comment by yuejunpeng** [comment]
@mapluisch Hello! It is true that you don’t need to resize the two images when you just concatenate them vertically, but I think the model does resize them to fit when feeding them to the clip image encoder.
